### PR TITLE
Error invalid basic msg

### LINKED ISSUE #513: Revisit ALL error messages / enums in zoracle
Expect to see the list of all error enums in `x/zoracle/internal/types/error.go` defined using `sdkerrors.Register`. All code in `zoracle` must be patched to use new error enums. Can re-use error from the standard std error types if applicable.

### PR DESCRIPTION
**Pull request by evilpeach**
This is a part of #513 

[Error Doc](https://docs.google.com/spreadsheets/d/1zh22ntVB21ZiqMDJEogwXUURKINyCf04SAX9IYpVgpc/edit#gid=1887575234)

### CODE REVIEW

File: chain/x/zoracle/internal/types/error.go [hunk 1]
```diff
@@ -1,13 +1,17 @@
...
+	"fmt"
+
 	sdk "github.com/cosmos/cosmos-sdk/types"
 )
 
 // DefaultCodespace is the Module Name
 const (
 	DefaultCodespace sdk.CodespaceType = ModuleName
 
+	CodeInvalidBasicMsg sdk.CodeType = 201
```
Comment (sorawit) [inline chain/x/zoracle/internal/types/error.go:14]: Add TODO: Change this to 101 after old errors were cleared.

File: chain/x/zoracle/internal/types/msgs.go [hunk 2]
```diff
@@ -51,25 +51,38 @@ func (msg MsgRequestData) Type() string { return "request" }
...
-		return sdk.ErrUnknownRequest("Sufficient validator count must be greater than zero")
+		return ErrInvalidBasicMsg(
+			"MsgRequestData: Sufficient validator count (%d) must be positive.",
+			msg.SufficientValidatorCount,
+		)
 	}
 	if msg.RequestedValidatorCount < msg.SufficientValidatorCount {
-		return sdk.ErrUnknownRequest("Request validator count must be greater than sufficient validator")
+		return ErrInvalidBasicMsg(
+			"MsgRequestData: Request validator count (%d) must be greater than sufficient validator count (%d).",
```
Comment (sorawit) [inline chain/x/zoracle/internal/types/msgs.go]: change to must not be less than. We accept the equality case.

File: chain/x/zoracle/internal/types/msgs.go [hunk 3]
```diff
@@ -114,13 +127,13 @@ func (msg MsgReportData) Type() string { return "report" }
 // ValidateBasic implements the sdk.Msg interface for MsgReportData.
 func (msg MsgReportData) ValidateBasic() sdk.Error {
 	if msg.RequestID <= 0 {
-		return sdk.ErrUnknownRequest("Request id must be greater than zero")
+		return ErrInvalidBasicMsg("MsgReportData: Request id (%d) must be positive.", msg.RequestID)
 	}
 	if msg.DataSet == nil || len(msg.DataSet) == 0 {
-		return sdk.ErrUnknownRequest("Data must not be empty struct")
+		return ErrInvalidBasicMsg("MsgReportData: DataSet must not be empty.")
```
Comment (sorawit) [inline chain/x/zoracle/internal/types/msgs.go]: Data set

**Review by sorawit — CHANGES_REQUESTED** [review]
So I guess we decide to use our errors in all places. That's fine.

**Review by sorawit — APPROVED** [review]

**Review by taobun — APPROVED** [review]

**Pull request merged**